ios/FluentUI/Command Bar/CommandBar.swift

```diff
@@ -65,20 +85,63 @@ open class CommandBar: UIView {
     public override func layoutSubviews() {
         super.layoutSubviews()
 
+        scrollView.contentInset = scrollViewContentInset()
+
+        leadingCommandGroupsStackView.layoutIfNeeded()
+        mainCommandGroupsStackView.layoutIfNeeded()
+        trailingCommandGroupsStackView.layoutIfNeeded()
+
         containerMaskLayer.frame = containerView.bounds
         updateShadow()
     }
 
+    /// Scrollable items shown in the center of the CommandBar
+    public var itemGroups: [CommandBarItemGroup] {
+        didSet {
+            mainCommandGroupsStackView.itemGroups = itemGroups
+        }
+    }
+
+    /// Items pinned to the leading end of the CommandBar
+    public var leadingItemGroups: [CommandBarItemGroup]? {
+        didSet {
+            guard let leadingItemGroups = leadingItemGroups else {
+                return
+            }
+
+            leadingCommandGroupsStackView.itemGroups = leadingItemGroups
+            constrainLeadingCommandGroupsStackView()
+        }
+    }
+
+    /// Items pinned to the trailing end of the CommandBar
+    public var trailingItemGroups: [CommandBarItemGroup]? {
+        didSet {
+            guard let trailingItemGroups = trailingItemGroups else {
+                return
+            }
+
+            trailingCommandGroupsStackView.itemGroups = trailingItemGroups
+            constrainTrailingCommandGroupsStackView()
+        }
+    }
+
     // MARK: - Private properties
 
-    private let itemGroups: [CommandBarItemGroup]
+    /// UIStackView holding the items pinned to the leading end of the CommandBar
+    private var leadingCommandGroupsStackView: CommandBarCommandGroupsStackView
 
-    private lazy var itemsToButtonsMap: [CommandBarItem: CommandBarButton] = {
-        let allButtons = itemGroups.flatMap({ $0 }).map({ button(forItem: $0) }) +
-            [leadingButton, trailingButton].compactMap({ $0 })
+    /// UIStackView holding the items in the middle of the CommandBar
+    private var mainCommandGroupsStackView: CommandBarCommandGroupsStackView
 
-        return Dictionary(uniqueKeysWithValues: allButtons.map { ($0.item, $0) })
-    }()
+    /// UIStackView holding the items pinned to the trailing end of the CommandBar
+    private var trailingCommandGroupsStackView: CommandBarCommandGroupsStackView
+
+    /// Leading constraint between the `containerView` and either the `leadingCommandGroupsStackView` or the parent view
+    private var leadingConstraint: NSLayoutConstraint?
+
+    /// Trailing constraint between the `containerView` and either the `trailingCommandGroupsStackView` or the parent view
+    private var trailingConstraint: NSLayoutConstraint?
```

Comment:
Maybe a bit late to be saying this, but would it be possible to just stick the leading, container, and trailing views in a horizontal stack and just hide the leading/trailing views when they're empty? Then you wouldn't need to dance around different constraints depending on if they're empty or not.